packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js

```diff
@@ -132,7 +132,29 @@ export function generateRichLineTooltipContent(d, timeFormatter, valueFormatter)
       '</tr>';
   });
   tooltip += '</tbody></table>';
+  return tooltip;
+}
 
+export function generateAreaChartTooltipContent(d, timeFormatter, valueFormatter) {
+  const total = d.series[d.series.length - 1].value;
+  let tooltip = '';
+  tooltip +=
+    "<table><thead><tr><td colspan='3'>" +
+    `<strong class='x-value'>${timeFormatter(d.value)}</strong>` +
```

Comment:
Should we sanitize / "html escape" what gets evaluated throughout this snipet?

Reply:
Is this what you mean @mistercrunch ?
https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-in-html
I wonder if we have a utility function for this.

Reply:
We use `dompurify.sanitize` in other places. Basically preventing XSS. In theory everything that is inputed by a user (webform) or comes from an external database should be sanitize when put into raw html... If `d.value` contains `<script>doBadStuff();</script>` it shouldn't get evaluated, it should just show up in the tooltip.
https://github.com/cure53/DOMPurify#what-does-it-do

Reply:
A bit more about this. React "sanitizes" ("html escapes") all strings you put into it, everywhere by default. If you want user input to get rendered as HTML for instance, you'll have to use `dangerouslySetInnerHTML` https://zhenyong.github.io/react/tips/dangerously-set-inner-html.html

I think Jinja too, unless you mark something as safe.

So anytime we see html strings getting concatenated with user-provided content, we need to watch out for XSS.

Reply:
Well explained! @mistercrunch